PIL-234 Add email verification steps

grilix opened the pull request:
Allow users to verify their email on the referrals flow. Email verification is required before referrals emails can be sent.

graszka22 commented on src/actions/userActions.js:
```diff
@@ -17,14 +17,34 @@
+  type: SENDING_OTP,
+});
+
+const otpPasswordSent = () => ({
```
all actions have "Action" suffix in the name, right? `otpPasswordSentAction` would be a better name

grilix replied:
Good catch, updated.

graszka22 commented on src/components/Verification/EmailVerified.js:
```diff
@@ -0,0 +1,44 @@
+  return (
+    <React.Fragment>
+      <MediumText>
+        You have confirmed your email!
```
Please add space between button and text. Also the text could be a bit bigger (on all verification screens).

<img width="410" alt="Screenshot 2020-02-19 at 11 15 28" src="https://user-images.githubusercontent.com/6280457/74825421-327c3280-530a-11ea-9000-2f25cf4b9e9b.png">

grilix replied:
Updated. Created a `Label` subcomponent so we have all of them look the same and we can easily update it. Not sure if there's already a component that we could have used for this (The `Typography/Label` doesn't look like a fit, let me know what you think.

graszka22 replied:
It's a shame that this branch is not updated with develop, because on develop we have those new Typography components and you'd just do something like this:
`<MediumText big>...</MediumText>`. And creating a new separate component for some text on one screen is a bit overkill. And the name is too generic as well. 
Here is what'd do to: looking at the designs, such "descriptions" are usually regular font with size 16sp (see for example the recovery flow, there are a lot of similar screens with description and a button). And in the design system such font is marked as just "Text". So I'd create a new component in the Typography file called "Text" or "RegularText" (so it follows the design system) and use that. How does it sound?

graszka22 commented on src/components/Verification/NoEmail.js:
```diff
@@ -0,0 +1,43 @@
+  return (
+    <React.Fragment>
+      <MediumText>
+        You don&apos;t have an email, please set your email first.
```
Add space here too

graszka22 commented on src/components/Verification/SendEmailCode.js:
```diff
@@ -0,0 +1,51 @@
+import { spacing } from 'utils/variables';
+
+const Label = styled(MediumText)`
+  padding-bottom: ${spacing.layoutSides}px;
```
`layoutSides` is for layout sides not bottom margin, use `spacing.mediumLarge`

graszka22 commented on src/screens/Users/VerifyEmail.js:
```diff
@@ -0,0 +1,182 @@
+  setEmail = () => {
+    const { navigation } = this.props;
+
+    navigation.navigate(ADD_EDIT_USER);
```
It's a bit weird:
- I press the go to profile button
- I'm brought to user settings
- I go back
- and I'm on the main user settings page instead of email verification

grilix replied:
You're right, fixed.

poocart commented on src/actions/userActions.js:
```diff
@@ -17,14 +17,34 @@
+  type: SENDING_OTP,
+});
+
+const otpPasswordSentAction = () => ({
```
As other actions have relation to oneTimePassword and only this has "otp" short handle:
```suggestion
const oneTimePasswordSentAction = () => ({
```

poocart commented on src/actions/userActions.js:
```diff
@@ -82,6 +109,43 @@ export type VerificationPhoneAction = {
+        },
+      });
+    } else {
+      dispatch(otpPasswordSentAction());
```
Can you explain the logic here? Why we mark it as sent only when error occurs?

grilix replied:
This is so we go back to the state we had before calling `dispatch(sendingOneTimePasswordAction());`. The state chain, on failure scenario would be: `sent` -> `sending` -> `sent`.

poocart commented on src/components/Verification/ConfirmCode.js:
```diff
@@ -0,0 +1,68 @@
+
+// utils
+import { spacing } from 'utils/variables';
+
```
```suggestion


```

grilix commented:
Thanks all for the comments! The UI for this feature changed a lot and it doesn't depend on the referrals anymore. So we extracted the code out and updated on #1701, please continue the thread there :heart:
